docs/src/model_agnostic_methods/conditional_feature_importance.rst

```diff
@@ -0,0 +1,116 @@
+other features, and a residual information term that is independent of the other features:
+
+.. math::
+    X_j = \nu_j(X_{-j}) + \epsilon_j, \quad \text{with} \quad \epsilon_j \perp\!\!\!\perp X_{-j} \text{ and } \mathbb{E}[\epsilon_j] = 0.
+
+Here :math:`\nu_j(X_{-j}) = \mathbb{E}[X_j | X_{-j}]` is the conditional expectation of
+:math:`X_j` given the other features. In practice, :math:`\nu_j` is unknown and has to be
+estimated from the data using a predictive model. 
+
+Then the perturbed feature :math:`X_j^p` is generated by keeping the predictable part
+:math:`\nu_j(X_{-j})` unchanged, and by replacing the residual :math:`\epsilon_j` by a
+randomly permuted version :math:`\epsilon_j^p`:
+
+.. math::
+    X_j^p = \nu_j(X_{-j}) + \epsilon_j^p, \quad \text{with} \quad \epsilon_j^p \sim \text{Perm}(\epsilon_j).
+
+
+.. note:: **Estimation of :math:`X_j^p`**
+
+    TODO: note on the fact estimating :math:`\nu_j` is supposed to be an easy task. 
+    Note on the fact that we can use any valid conditional sampler can be used.
+
+
+Inference
+---------
+TODO finish the section on inference
+
+
+Regression example
+------------------
+The following example illustrates the use of CFI on a regression task with::
+
+    >>> from sklearn.datasets import make_regression
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.model_selection import train_test_split
+    >>> from hidimstat import CFI
+
+
+    >>> X, y = make_regression(n_features=2)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y)
+    >>> model = LinearRegression().fit(X_train, y_train)
+    
+    >>> cfi = CFI(estimator=model, imputation_model_continuous=LinearRegression())
+    >>> cfi.fit(X_train, y_train)
+    >>> features_importance = cfi.importance(X_test, y_test)
+
+
+Classification example
+----------------------
+To measure feature importance in a classification task, a classification loss should be
+used, in addition, the prediction method of the estimator should output the corresponding 
+type of prediction (probabilities or classes). The following example illustrates the use
+of CFI on a classification task with::
+
+    >>> from sklearn.datasets import make_classification
+    >>> from sklearn.ensemble import RandomForestClassifier
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.metrics import log_loss
+    >>> from sklearn.model_selection import train_test_split
+    >>> from hidimstat import CFI
+
+    >>> X, y = make_classification(n_features=4)
+    >>> X_train, X_test, y_train, y_test = train_test_split(X, y)
+    >>> model = RandomForestClassifier().fit(X_train, y_train)
+    >>> cfi = CFI(
+    ...     estimator=model,
+    ...     imputation_model_continuous=LinearRegression(),
+    ...     loss=log_loss,
+    ...     method="predict_proba",
+    ... )
+    >>> cfi.fit(X_train, y_train)
+    >>> features_importance = cfi.importance(X_test, y_test)
```

Comment:
I added snippets for minima examples of usage. They're heavier than in classical sklearn snippets. Let me know if you think it's too long.

Reply:
thx